### PR TITLE
[BOUNTY] Add Phi-2

### PR DESCRIPTION
**Pull request by JushBJJ**
**Issues that need to be addressed:** https://github.com/tenstorrent/tt-buda/issues/42
**Video of it working:** https://www.youtube.com/watch?v=a3gCi8B87zA
**Card:** Grayskull e75

### Requirements
- Transformers >=4.42.0
- ~40GB+ depending on how long the generation is

Bounty: https://github.com/tenstorrent/tt-buda-demos/issues/21

### CODE REVIEW

**Comment by milank94** [comment]
Awesome @JushBJJ -- same comments as https://github.com/tenstorrent/tt-buda-demos/pull/37 on the formatting of the demo.

**Comment by JushBJJ** [comment]
I'll add it to Phi after I rewrite Qwen's file
https://github.com/tenstorrent/tt-buda/issues/42#issuecomment-2258909588

**Comment by JushBJJ** [comment]
This should be good now @milank94 for final review and testing. Although note that Phi2 is incredibly slow on an e75 and RAM usage is massive, when only 9GB is used initially the usage can go up to 50GB+ depending on how long the generation takes.

**Comment by milank94** [comment]
@JushBJJ thank you for your patience as we're close to releasing a new release of pybuda with the changes to support this model demo. Can you switch the target for this PR to be `track_pybuda`? https://github.com/tenstorrent/tt-buda-demos/tree/track_pybuda

The process flow is that we use `track_pybuda` in our internal CI to test all of the demos and changes, then we merge those into `main`.

I'll add a similar comment to your other PR: https://github.com/tenstorrent/tt-buda-demos/pull/37

File: model_demos/tests/test_pytorch_phi2.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+
+from nlp_demos.phi2.pytorch_phi2_text_generation import run_phi2_causal_lm
+
+@pytest.mark.qwen1_5
+def test_qwen1_5_causal_lm_pytorch(clear_pybuda, test_device, batch_size):
```
Comment (milank94) [inline model_demos/tests/test_pytorch_phi2.py:9]:
```suggestion
def test_phi2_causal_lm_pytorch(clear_pybuda, test_device, batch_size):
```

**Comment by milank94** [comment]
Hey @JushBJJ can you switch the target branch to be: [mkordic/rc_20240830](https://github.com/tenstorrent/tt-buda-demos/tree/mkordic/rc_20240830)?

**Comment by JushBJJ** [comment]
Done

**Pull request merged**